HPXMLtoOpenStudio/resources/utility_bills.rb

```diff
@@ -105,47 +47,39 @@ def self.marginal_rate_to_average_rate(marginal_rate, fixed_charge, household_co
     return marginal_rate + 12.0 * fixed_charge / household_consumption
   end
 
-  def self.get_state_code_to_padd
-    # https://www.eia.gov/tools/glossary/index.php?id=petroleum%20administration%20for%20defense%20district
-    padd_to_state_codes = { 'PADD 1A' => ['CT', 'MA', 'ME', 'NH', 'RI', 'VT'],
-                            'PADD 1B' => ['DE', 'DC', 'MD', 'NJ', 'NY', 'PA'],
-                            'PADD 1C' => ['FL', 'GA', 'NC', 'SC', 'WV', 'VA'],
-                            'PADD 2' => ['IA', 'IL', 'IN', 'KS', 'KY', 'MI', 'MN', 'MO', 'ND', 'NE', 'OH', 'OK', 'SD', 'TN', 'WI'],
-                            'PADD 3' => ['AL', 'AR', 'LA', 'MS', 'NM', 'TX'],
-                            'PADD 4' => ['CO', 'ID', 'MT', 'UT', 'WY'],
-                            'PADD 5' => ['AK', 'AZ', 'CA', 'HI', 'NV', 'OR', 'WA'] }
+  def self.get_eia_seds_rate(state_code, fuel_type)
+    msn_code_map = {
+      HPXML::FuelTypeElectricity => 'ESRCD',
+      HPXML::FuelTypeNaturalGas => 'NGRCD',
+      HPXML::FuelTypeOil => 'DFRCD',
+      HPXML::FuelTypePropane => 'PQRCD',
+      HPXML::FuelTypeCoal => 'CLRCD',
+      HPXML::FuelTypeWoodCord => 'WDRCD',
+      HPXML::FuelTypeWoodPellets => 'WDRCD'
+    }
+    unit_conv = {
+      HPXML::FuelTypeElectricity => 0.003412, # convert $/MMBtu to $/kWh
+      HPXML::FuelTypeNaturalGas => 0.1, # convert $/MMBtu to $/therms
+      HPXML::FuelTypeOil => 0.139, # convert $/MMBtu to $/gallons of oil
+      HPXML::FuelTypePropane => 0.0916, # convert $/MMBtu to $/gallons of propane
+      HPXML::FuelTypeCoal => 1, # $/MMBtu
+      HPXML::FuelTypeWoodCord => 1, # $/MMBtu
+      HPXML::FuelTypeWoodPellets => 1 # $/MMBtu
+    }
+
+    rows = CSV.read(File.join(File.dirname(__FILE__), '../../ReportUtilityBills/resources/simple_rates/pr_all_update.csv'))
+    rows.each do |row|
+      next if row[1].upcase != state_code.upcase # State
+      next if row[2].upcase != msn_code_map[fuel_type] # EIA SEDS MSN code
 
-    state_code_to_padd = {}
-    padd_to_state_codes.each do |padd, state_codes|
-      state_codes.each do |state_code|
-        state_code_to_padd[state_code] = padd
+      if fuel_type == HPXML::FuelTypeCoal
+        seds_rate = row[40] # Use 2007 prices for coal. For 2008 forward, EIA assumes there is zero residential sector coal consumption in the United States, and SEDS does not estimate a price.
+      else
+        seds_rate = row[-1]
+        seds_rate = row[-2] if seds_rate.nil? # If the rate for the latest year is unavailable, use the rate from the previous year.
       end
```

Comment:
When we update the prices in the future, A) the `[40]` for coal may no longer be correct and B) we have to look back more than a single year (`[-2]`) for a non-nil value. Please use more generic, future-proof logic. For example, simply return the last non-nil value for any fuel type.